go-controller/pkg/ovn/obj_retry.go

```diff
@@ -0,0 +1,139 @@
+package ovn
+
+import (
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const retryObjInterval = 30 * time.Second
+
+// retryObjEntry is a generic object caching with retry mechanism
+//that resources can use to eventually complete their intended operations.
+type retryObjEntry struct {
+	newObj     interface{}
+	oldObj     interface{}
+	config     interface{}
+	timeStamp  time.Time
+	backoffSec time.Duration
+	// whether to include this object in the retry iterations
+	ignore bool
+}
+
+// initRetryObjWithAdd tracks an object that failed to be created to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObjWithAdd(obj interface{}, key string) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.newObj = obj
+	} else {
+		r.entries[key] = &retryObjEntry{newObj: obj,
+			timeStamp: time.Now(), backoffSec: 1, ignore: true}
+	}
+}
+
+// initRetryWithDelete tracks an object that failed to be deleted to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObjWithDelete(obj interface{}, key string, config interface{}) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.oldObj = obj
+		if entry.config == nil {
```

Comment:
haven't had a chance to check the namespace.go and policy.go in depth, but we are 100% sure we don't want to overwrite an existing config unlike oldObj right? like I understand why this check is here because we call config=nil from most of the places, just wondering if 
```
oc.retryNetPolices.initRetryObjWithDelete(np.policy, key, np)
```
is a one time thing? I guess answer is yes.

Reply:
u mean one time for every delete event right ? if so then yes